redaxo/src/core/lib/util/i18n.php

```diff
@@ -148,6 +145,44 @@ private static function getMsg($key, $htmlspecialchars, array $args)
         return $msg;
     }
 
+    /**
+     * Returns the message fallback for a missing key in main locale.
+     *
+     * @param string $key
+     * @param array  $args
+     *
+     * @return string
+     */
+    private static function getMsgFallback($key, array $args)
+    {
+        $fallback = "[translate:$key]";
+
+        $msg = rex_extension::registerPoint(new rex_extension_point('I18N_MISSING_TRANSLATION', $fallback, [
+            'key' => $key,
+            'args' => $args,
+        ]));
+
+        if ($msg !== $fallback) {
+            return $msg;
+        }
+
+        foreach (rex::getProperty('lang_fallback') as $locale) {
+            if (self::$locale === $locale) {
+                continue;
+            }
+
+            if (empty(self::$loaded[$locale])) {
```

Comment:
ggf. sollte man mal testen, was die performance dazu sagt, wenn hier zusätzliche sprachkataloge geladen werden müssen (es müsste ja ausreichen, wenn nur 1 key in der aktuellen sprache fehlt, dass dann noch EN geladen wird...?)

Reply:
Ja, ich hatte vor der Umsetzung mir das Backend nochmal in Blackfire angeschaut, da ich noch in Erinnerung habe, dass i18n da einen wesentlichen Anteil hat.
https://blackfire.io/profiles/6d45756d-8ddd-4dbb-a52a-376a12dc3a7e/graph
Es ist zwar ein wesentlicher Anteil, aber trotzdem so niedrig, dass ich das als unproblematisch werten würde.
Ansonsten hätte ich einen Cache für rex_i18n gebaut, ähnlich wie alle package.yml gemeinsam gecacht werden. Aber ich hab den Eindruck, ist eigentlich nicht wirklich notwendig, oder?

Reply:
Die Yml-Dateien hatten ja einen viel größeren Einfluss.
Sobald man markitup aktiviert, was 2 Yaml-Dateien lädt, geht die Zeit hoch von 50ms auf 322ms:
https://blackfire.io/profiles/eb45aa9a-7c27-4c9f-a8a4-866f608bcc56/graph

Reply:
ok, cool!

Reply:
aber markitup sollten wir dann auch versuchen nochmal anzuschauen.. ;)